### ISSUE TITLE
Create Metadata Generation web service

### ISSUE THREAD

**Issue by lossyrob** [hotosm/oam-server#1]
This web service listens for requests that are sent when a HOT OAM user submits new imagery. This will kick off a process that runs [oin-meta-generator](https://github.com/openimagerynetwork/oin-meta-generator) against the new imagery, thus allowing it to participate in OIN.

Outstanding questions:
- Should we be doing validation at this point?
- Do we need to notify OAM of the new imagery?

**Comment by cgiovando** [comment]
> Outstanding questions:
> - Should we be doing validation at this point?  

Yes, we need validation for OIN and OAM metadata. It should be handled through API. The response in case of exception can either be "not compliant" or "missing".  

> - Do we need to notify OAM of the new imagery?  

The service can either add the resource to the OIN index which is then indexed by OAM, or update both OIN and OAM. If we only notify OIN, there may be a delay until the resource shows up in OAM. Another option is to add it to the OIN index and then actively trigger the OAM refresh service.  

The same metadata generation service should run when a processing task (imagery to tile service) is submitted. The user should be prompted to add any tile service specific elements that are not already passed through from the OIN source metadata. In the case someone uploads a new imagery dataset and at the same time of upload requests the OAM tile service creation, then both OIN metadata and tile service specific metadata should be created at the same time.

**Comment by lossyrob** [comment]
Validation logic should be handled by oin-meta-generator; part of this task can be to add the validation feature to that project. We should define what valid and invalid imagery is and write that issue into oin-meta-generator.

**Comment by lossyrob** [comment]
> The service can either add the resource to the OIN index which is then indexed by OAM, or update both >OIN and OAM. If we only notify OIN, there may be a delay until the resource shows up in OAM. Another >option is to add it to the OIN index and then actively trigger the OAM refresh service.

I would recommend having a simple service call to notify OAM of the new imagery.

**Comment by lossyrob** [comment]
> The same metadata generation service should run when a processing task (imagery to tile service) is >submitted. The user should be prompted to add any tile service specific elements that are not already >passed through from the OIN source metadata. In the case someone uploads a new imagery dataset >and at the same time of upload requests the OAM tile service creation, then both OIN metadata and tile >service specific metadata should be created at the same time.

Since the metadata generation service and tms generation service will be independent services, both can be called as part of an upload.

What you describe seems to be more of a UI workflow element; which is not part of the service, and as far as I understand it not part of oam-server. Where will the project that hosts the user upload UI and client application be located?

**Comment by cgiovando** [comment]
User will be able to upload through a Web page on OAM or any other client via API. But since OIN is intended as a static repository without an actual service running, where do see oin-meta-generator be hosted?
